docs/checkit.js

```diff
@@ -685,30 +685,43 @@ CheckIt.prototype.deleteCalendar = function() {
     
     // Guard against accidental clicks of the delete button
     
+
     var confirmation = confirm("Are you sure you want to delete your calendar?");
-    if (confirmation) {
-        
-        var checkit = this;
-                
-        return checkit.store.getActive().then(function(currentCalendarId) {
+    if (!confirmation) {
+      return Promise.reject("User cancelled the delete operation.");
+    }
+
+    var checkit = this;
+
+    return checkit.store.getActive()
+        .catch(function(err) {
+            console.error("Couldn't get active calendar id: " + err);
+        })
+        .then(function(currentCalendarId) {
             // remove calendar from dropdown
             checkit.removeFromCalendarDropdown(currentCalendarId,
-            checkit.$calendarDropdown);
+                                               checkit.$calendarDropdown);
+
             // remove calendar state from storage
-            return checkit.store.removeById(currentCalendarId).catch(function(err) {
-                console.log("Unable to remove calendar from storage.");
-                return err;
+            return checkit.store.removeById(currentCalendarId)
+                .catch(function(err) {
+                    console.error("Unable to remove calendar from storage.");
+                    // TODO: yes this is an error, but what do we want to do now?
+                    // Do we want to plow onward and clear the div or is this worth
+                    // stopping the entire app over?
```

Comment:
So what you'd really like is for errors that can recover gracefully, but not if that means that we misrepresent things to the user. We wouldn't want to say we deleted the calendar, for example, and then have the user refresh the page and discover that we didn't. This could happen if e.g. their internet connection was unreliable and the call to remove failed.

Perhaps most importantly, we don't want to end up in a state where the database doesn't make sense anymore. As long as we avoid that, then it's not *that* big of a deal.

Do you have a way to [tell the user](https://notifyjs.com/) that there was an error of some kind?